test/callbacks.jl

```diff
@@ -0,0 +1,63 @@
+#  Copyright 2017, Iain Dunning, Joey Huchette, Miles Lubin, and contributors
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+using JuMP
+using Test
+
+struct DummyCallbackData end
+
+@testset "LazyConstraint" begin
+    mock = MOI.Utilities.MockOptimizer(
+        MOI.Utilities.UniversalFallback(MOI.Utilities.Model{Float64}())
+    )
+    model = direct_model(mock)
+    @variable(model, 0 <= x <= 2.5, Int)
+    con = @build_constraint(x <= 2)
+    c = MOI.submit(model, MOI.LazyConstraint(DummyCallbackData()), con)
```

Comment:
Isn't submit not supposed to return anything ? If we make it return nothing in MOI then it might break JuMP tests